src/Traits/VerifiableTrait.php

```diff
@@ -15,20 +14,33 @@ trait VerifiableTrait
 
     protected function loadActiveVerifications(string $action)
     {
-        $newVerifications = $this->shouldKeepDuringSession($action)
-            ? VerificationCode::allFor($this)->whereNull('verifies_until')->whereNotNull('used_at')->get()
-            : VerificationCode::allFor($this)->where('verifies_until', '>', Carbon::now()->toDateTime())->get();
+        $newVerifications = $this->getNewVerifications($action);
 
         $this->activeVerifications = is_null($this->activeVerifications)
             ? $newVerifications
             : $this->activeVerifications->merge($newVerifications);
     }
 
-    private function shouldKeepDuringSession(string $action): bool
+    private function getNewVerifications(string $action): Collection
     {
-        return (Config::get('verifications.actions.'.$action.'.keep_verified_during_session') === true)
-            && Session::has('last_activity')
-            && (Session::get('last_activity') < Carbon::now()->addMinutes(Config::get('session.lifetime'))->toDateTime());
+        $now = Carbon::now();
+
+        switch (Config::get('verifications.actions.'. $action .'.expires_from')) {
+            case 'last-activity':
+                 $verificationsQuery = VerificationCode::allActiveForAction($this, $action, $now->toDateTime());
```

Comment:
Spravil si fasadu na VerificationCodesRepository? Riskujeme tym, ze to niekto bude pozuivat a pritom by nemal  nikto. To je low-level vec package. Ci islo ti o to, ze nevies ako to injectnut do traitu, preto si to takto obisiel?

Reply:
Toto je len scope v ramci modelu **VerificationCode**, chcel som sa vyhnut spagetovemu kodu, tak mi to prislo rozumne.

Reply:
Jaaaj, chapem, ja by som tam doplnil `query()`, nech je to zrozumitelnejsie ze je to scope nad query buildrom. Nebolo mi to jasne:
```
$verificationsQuery = VerificationCode::query()->allActiveForAction($this, $action, $now->toDateTime());
```